src/components/manager.js

```diff
@@ -148,6 +148,9 @@ export class Manager extends Component {
   componentWillUnmount() {
     this._detachEvents();
   }
+
+  viewedIndexes = new Set();
```

Comment:
`viewedIndexes` is just a property on the class because we don’t need to re-render the Manager when it changes. Therefore having it in state isn't necessary.